### PR TITLE
Fix bug in metric computation when there is overlap with grid objective

### PR DESCRIPTION
**Pull request by desilinguist**
This PR addresses #564. 

- Moves the overlap checking from the configuration level (in `config.py`) to the experiment level (in `learner.py`). 
- Adds another check for code added in the earlier metric computation PR: the message about probabilities being used to compute the metric is now _not_ shown when computing the objective since doing so might send conflicting messages to the user given the overlap message.
- Remove old test for overlap checking from `test_input.py`
- Add new test for overlap checking to `test_classification.py`.

### CODE REVIEW

File: tests/test_classification.py [hunk 1]
```diff
@@ -199,8 +204,8 @@ def check_label_index_order_with_pos_label_str(use_api=True):
         # if we are using the API, create a learner instance directly
         # otherwise run a configuration file to train a learner
         if use_api:
-            clf = Learner('LogisticRegression', pos_label_str=pos_label_str)
-            clf.train(train_fs, grid_search=False)
+        clf = Learner('LogisticRegression', pos_label_str=pos_label_str)
```
Comment (bndgyawali) [inline tests/test_classification.py]: should not this be indented?

File: tests/test_classification.py [hunk 2]
```diff
@@ -311,7 +316,7 @@ def check_binary_predictions_for_pos_label_str(label_list,
         clf_skll = Learner('LogisticRegression',
                            pos_label_str=pos_label_str,
                            probability=probability)
-        clf_skll.train(train_fs, grid_search=False)
+    clf_skll.train(train_fs, grid_search=False)
```
Comment (bndgyawali) [inline tests/test_classification.py]: this too.

File: tests/test_classification.py [hunk 3]
```diff
@@ -356,11 +361,11 @@ def test_binary_predictions_for_pos_label_str():
          use_pos_label_str,
          probability,
          use_api) in product([['B', 'C'],
-                              [1, 2],
-                              ['FRAG', 'NONE']],
+                                  [1, 2],
```
Comment (bndgyawali) [inline tests/test_classification.py]: this too.

File: tests/test_classification.py (hunk 1 above)
Comment (desilinguist) [reply tests/test_classification.py]: Weird, not sure how that got unindented.

File: tests/test_classification.py [hunk 4]
```diff
@@ -1668,3 +1673,69 @@ def test_metric_values_for_classification():
...
+                                         values_to_fill_dict,
+                                         'metrics_objective_overlap',
+                                         good_probability_option=True)
+
+    # run this configuration file and get the output JSON
+    # files for each experiment
+    results_json_paths = run_configuration(config_path, local=True, quiet=True)
+
+    # make a dummy objective if we do not have any for the
+    # purposes if iteration
```
Comment (aoifecahill) [inline tests/test_classification.py]:
```suggestion
    # purposes of iteration
```

**Pull request merged**